### ISSUE TITLE
element() function does not interpolate 'aws_elasticache_cluster.bar.cache_nodes.*.address'

### ISSUE THREAD

**Issue by samueljamesmarshall** [hashicorp/terraform#8956]
### Terraform Version

0.7.1
### Affected Resource(s)
- aws_elasticache_cluster
### Terraform Configuration Files

``` hcl
variable "cache_nodes" {
  default = 3
}

resource "aws_elasticache_cluster" "memcached_elasticache" {
  ...
  num_cache_nodes = "${var.cache_nodes}"
  ...
}

resource "aws_route53_record" "memcached_elasticache_node" {
   ...
   records = ["${element(aws_elasticache_cluster.memcached_elasticache.cache_nodes.*.address, count.index)}"]
   count = "${var.cache_nodes}"
   ...
}
```
### Expected Behavior

The route53 resource should set a record for each cache node e.g. "memcached.icxg4p.0001.apse2.cache.amazonaws.com", "memcached.icxg4p.0002.apse2.cache.amazonaws.com", "memcached.icxg4p.0003.apse2.cache.amazonaws.com".
### Actual Behavior

```
Error running plan: 3 error(s) occurred:

* At column 3, line 1: element: argument 1 should be type list, got type string in:

${element(aws_elasticache_cluster.memcached_elasticache.cache_nodes.*.address, count.index)}
* At column 3, line 1: element: argument 1 should be type list, got type string in:

${element(aws_elasticache_cluster.memcached_elasticache.cache_nodes.*.address, count.index)}
* At column 3, line 1: element: argument 1 should be type list, got type string in:

${element(aws_elasticache_cluster.memcached_elasticache.cache_nodes.*.address, count.index)}
```
### Workaround

``` hcl
resource "aws_route53_record" "memcached_elasticache_node" {
   ...
   records = ["memcached.${element(split(".", aws_elasticache_cluster.memcached_elasticache.configuration_endpoint), 1)}.000${count.index + 1}.${element(split(".", aws_elasticache_cluster.memcached_elasticache.configuration_endpoint), 3)}.cache.amazonaws.com"]
   count = "${var.cache_nodes}"
   ...
}
```
### Steps to Reproduce
1. `terraform plan`

**Comment by jim3ma** [comment]
This is a workaround just for redis, memcached may be similar.

```
count = "${var.blabal}"
record = "${aws_elasticache_cluster.default.cluster_id}.${element(split(".", aws_elasticache_cluster.default.cache_nodes.0.address), 1)}.${format("%04d", count.index + 1)}.${element(split(".", aws_elasticache_cluster.default.cache_nodes.0.address), 3)}.cache.amazonaws.com"
```

**Comment by mitchellh** [comment]
Fixed with #10657.
